### PR TITLE
Fix histogram data point comment about supported boundaries

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes: https://github.com/open-telemetry/opentelemetry-proto/issues/259

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -432,6 +432,12 @@ message IntHistogramDataPoint {
   // buckets if a histogram is provided.
   sfixed64 sum = 5;
 
+  // A histogram may optionally contain the distribution of the values in the
+  // population. In that case one the "explicit_bounds" and "bucket_counts"
+  // fields must be defined. Otherwise both fields must be omitted in which case
+  // the distribution of values in the histogram is unknown and only the total
+  // count and sum are known.
```
Comment (jsuereth) [inline opentelemetry/proto/metrics/v1/metrics.proto:439]: Nit: This is a bit hard to parse.

Proposal:
```
If the histogram contains a distribution of values then one of the "explicit_bounds" or 
"bucket counts" fields must be defined.   
If the histogram is not a distribution of values, then both "explicit_bounds" and "bucket_counts"
must be omitted and only "count" and "sum" are known.
```

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 2]
```diff
@@ -508,6 +506,12 @@ message HistogramDataPoint {
   // buckets if a histogram is provided.
   double sum = 5;
 
+  // A histogram may optionally contain the distribution of the values in the
```
Comment (jsuereth) [inline opentelemetry/proto/metrics/v1/metrics.proto:509]: Same comment as above.

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 3]
```diff
@@ -432,6 +432,12 @@ message IntHistogramDataPoint {
   // buckets if a histogram is provided.
   sfixed64 sum = 5;
 
+  // A histogram may optionally contain the distribution of the values in the
```
Comment (jsuereth) [inline opentelemetry/proto/metrics/v1/metrics.proto:435]: Does this belong as a message-level comment?

**Review by jsuereth — CHANGES_REQUESTED** [review]

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 1 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: Realized that the message was also wrong, we need both.

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 2 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: done.

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 3 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: done.

**Review by jsuereth — APPROVED** [review]

**Review by jmacd — APPROVED** [review]
